### ISSUE TITLE
Changes made using Atom not saved to Anki

### ISSUE THREAD

**Issue by srb310** [pointtonull/anki-external_editor#8]
When I use external editor to edit html of cards in Atom, it does not save changes to the Anki card. The temp files are saved but the changes are not reflected in Anki. Atom saves all my other files (outside of Anki) properly. It works fine for other editors like Notepad++

**Comment by pointtonull** [comment]
Hi!

Sorry for the long delay to reply. I installed atom in Linux and I was able to reproduce the problem. And then, to find a solution.

What happens is that Atom creates a sub-process that creates the editor window with your text; and immediately created this windows it ends the original process. Anki cannot know what happens inside Atom, so it assumes the job is done when Atom (the original process) is done. But is not, because Atom delegated it to the Editor window.

Long story short. You can change this behavior by adding the `--foreground` flag. If this does not work in Windows try adding the `--wait` flag. Or any combination of these two.

e.g.:
```
{
    "editor": "atom --foreground"
}
```

I will release an update with the new behavior fixed for when Atom is automatically detected.

Thank you for your report.

**Comment by srb310** [comment]
Putting the `--foreground` flag fixed the issue. Thank you so much for helping and for creating the addon!

**Comment by warfarincoumadin** [comment]
@pointtonull 
There is a related issue in Visual Studio Code.

Saving and clozing the tab in VSC will not export the changes to Anki, VSC has to be closed completely. Is there any way around this? So that saving the tab alone will update the anki note without the need to close the program?

**Comment by dempe** [comment]
Note for those using Vim or MacVim, the relevant option is `--nofork`.

**Comment by ryerrabelli** [comment]
Has anyone also had this problem that couldn't be fixed by the above options? I did and I didn't see this thread before, so I made a code change to make it work. If there are still people who have this issue, then I'll consider making a pull request with my fix. Fyi, I'm on a Mac. And it wasn't saving on either textedit or atom for me.
